src/cryptojwt/key_bundle.py

```diff
@@ -551,32 +561,34 @@ def get(self, typ="", only_active=True):
             otherwise the appropriate keys in a list
         """
         self._uptodate()
-        _typs = [typ.lower(), typ.upper()]
 
-        if typ:
-            _keys = [k for k in self._keys if k.kty in _typs]
-        else:
-            _keys = self._keys
+        with self._lock_reader:
+            if typ:
+                _typs = [typ.lower(), typ.upper()]
+                _keys = [k for k in self._keys if k.kty in _typs]
+            else:
+                _keys = self._keys
 
         if only_active:
             return [k for k in _keys if not k.inactive_since]
 
         return _keys
 
-    def keys(self):
+    def keys(self, update: bool = True):
         """
         Return all keys after having updated them
 
         :return: List of all keys
         """
-        self._uptodate()
-
-        return self._keys
+        if update:
+            self._uptodate()
+        with self._lock_reader:
+            return self._keys
```

Comment:
Need a shallow copy here. Optimization, save the copy and remove copy every time keys_writer() is called.

Reply:
670d0bc087c9adfa4ad945f68f80921202d5e138